lxd/cgroup/init.go

```diff
@@ -64,6 +64,204 @@ func (info *Info) Mode() string {
 	return "unknown"
 }
 
+// Resources is a generic type used to abstract resource control features
+// support for the legacy and unified hierarchy.
+type Resources int
+
+const (
+	// Blkio resource control
+	Blkio Resources = iota
+	// BlkioWeight resource control
+	BlkioWeight
+	// CPU resource control
+	CPU
+	// CPUAcct resource control
+	CPUAcct
+	// CPUSet resource control
+	CPUSet
+	// Devices resource control
+	Devices
+	// Freezer resource control
+	Freezer
+	// Memory resource control
+	Memory
+	// MemoryMaxUsage resource control
+	MemoryMaxUsage
+	// MemorySwap resource control
+	MemorySwap
+	// MemorySwapMaxUsage resource control
+	MemorySwapMaxUsage
+	// MemorySwapUsage resource control
+	MemorySwapUsage
+	// MemorySwappiness resource control
+	MemorySwappiness
+	// NetPrio resource control
+	NetPrio
+	// Pids resource control
+	Pids
+)
+
+// SupportsVersion indicates whether or not a given cgroup control knob is
+// available and in which type of cgroup filesystem.
+func (info *Info) SupportsVersion(resource Resources) (Backend, bool) {
+	switch resource {
+	case Blkio:
+		val, ok := cgControllers["blkio"]
+		if ok && val == V1 {
+			return V1, ok
+		}
+		return Unavailable, false
```

Comment:
Line break before this and all subsequent similar returns please (to split the block which handles the retrieval+check from the map from the rest of the logic).